shrink-path: added glob and quote options

mjklemm opened the pull request:
The new option adds an asterix to the uniquely shortened name, such
that, if copy'n'pasted, zsh globbing will expand to the full path name
of the shortened path:

shrink-path -g:
"/net/software/modules/modulefiles" -> "/n*/s*/modules/m*"

shrink-path -l -g:
"/net/software/modules/modulefiles" -> "/n*/s*/modules/modulefiles"

flatiron32 commented on plugins/shrink-path/README.md:
````diff
@@ -37,6 +38,7 @@ The following options are available:
 
 ```
     -f, --fish       fish simulation, equivalent to -l -s -t.
+    -g, --global     Add asterix to allow globbing of shrunk path.
````
This appears to be a typo.
```suggestion
    -g, --glob     Add asterix to allow globbing of shrunk path.
```

mjklemm replied:
Thanks for catching that!  I have committed your suggestion.

mcornella commented:
Hi @mjklemm, now that #7382 has been merged, can you make `-g` a special case where the ellipsis character (`-e`) is an asterisk?

mjklemm commented:
> Hi @mjklemm, now that #7382 has been merged, can you make `-g` a special case where the ellipsis character (`-e`) is an asterisk?

Done.  Actually having pull request #7382 made things a lot easier and reduced this PR to a four-liner or so.

jeffmhubbard commented:
Asterix? Did you mean 'asterisk'?

mjklemm commented:
One thing that I found during testing, -g does not handle whitespace and special symbols in the shortened path name.  `~/tmp/bla blubs/foo` becomes `~/tm*/bla blu*/foo*` w/o adding a backslash to the whitespace to escape it. Not sure how to fix that, yet.

mjklemm commented:
> Asterix? Did you mean 'asterisk'?

Glad that I did not write "Obelix"...  Thanks for catching the typo.

mcornella commented:
> One thing that I found during testing, -g does not handle whitespace and special symbols in the shortened path name. `~/tmp/bla blubs/foo` becomes `~/tm*/bla blu*/foo*` w/o adding a backslash to the whitespace to escape it. Not sure how to fix that, yet.

Is that a consequence of using the new code or it was already a problem with the previous version of the PR?

mjklemm commented:
> > One thing that I found during testing, -g does not handle whitespace and special symbols in the shortened path name. `~/tmp/bla blubs/foo` becomes `~/tm*/bla blu*/foo*` w/o adding a backslash to the whitespace to escape it. Not sure how to fix that, yet.
> 
> Is that a consequence of using the new code or it was already a problem with the previous version of the PR?

That's an issue that is already present in the current code of shrink_path:

```
sedna ~/tm/bla blu/foo [0:0]> pwd
/net/home/mklemm/tmp/bla blubs/foo
sedna ~/tm/bla blu/foo [0:0]> shrink_path -t -l
~/tm/bla blu/foo
```

(Same without -l and -t)

mcornella commented:
You can use `${(q)var}` to quote special characters, see:
```
$ quotethis="spaces and special characters like ?[]{}"
$ echo ${(q)quotethis}
spaces\ and\ special\ characters\ like\ \?\[\]\{\}
```

mjklemm commented:
> You can use `${(q)var}` to quote special characters, see:
> 
> ```
> $ quotethis="spaces and special characters like ?[]{}"
> $ echo ${(q)quotethis}
> spaces\ and\ special\ characters\ like\ \?\[\]\{\}
> ```

Thanks for the useful hint!  I have added "-q" to do the quoting.  I have also updated the README to show the effect of quoting the path.
